php-frontend/src/main/java/org/sonar/php/parser/PHPLexicalGrammar.java

```diff
@@ -0,0 +1,454 @@
+  CLASS_MEMBER_ACCESS,
+  OBJECT_MEMBER_ACCESS,
+  SIMPLE_INDIRECT_REFERENCE,
+  STATIC_MEMBER,
+  COMPOUND_VARIABLE,
+  CLASS_NAME,
+  FUNCTION_CALL_PARAMETER_LIST,
+  PARAMETER_LIST_FOR_CALL,
+  DIMENSIONAL_OFFSET,
+  STATIC_SCALAR,
+  OBJECT_DIM_LIST,
+  PARENTHESIS_EXPRESSION,
+  YIELD_EXPRESSION,
+  COMBINED_SCALAR,
+  COMMON_SCALAR,
+  BOOLEAN_LITERAL,
+  LEXICAL_VARS,
+  LEXICAL_VAR_LIST,
+  LEXICAL_VAR,
+  LOGICAL_XOR_EXPR,
+  LOGICAL_OR_EXPR,
+  LOGICAL_OR_OPERATOR,
+  BITEWISE_AND_EXPR,
+  BITEWISE_XOR_EXPR,
+  BITEWISE_OR_EXPR,
+  LOGICAL_AND_EXPR,
+  LOGICAL_AND_OPERATOR,
+  CONDITIONAL_EXPR,
+  ASSIGNMENT_EXPR,
+  MULTIPLICATIVE_EXPR,
+  MULIPLICATIVE_OPERATOR,
+  ADDITIVE_EXPR,
+  ADDITIVE_OPERATOR,
+  SHIFT_EXPR,
+  SHIFT_OPERATOR,
+  RELATIONAL_EXPR,
+  RELATIONAL_OPERATOR,
+  EQUALITY_EXPR,
+  EQUALITY_OPERATOR,
+  CONCATENATION_EXPR,
+  POSTFIX_EXPR,
+  UNARY_EXPR,
+  ASSIGNMENT_BY_REFERENCE,
+  ASSIGNMENT_OPERATOR,
+  COMPOUND_ASSIGNMENT,
+  CAST_TYPE,
+  LOGICAL_ASSIGNMENT,
+  INTERNAL_FUNCTION,
+  NEW_EXPR,
+  COMBINED_SCALAR_OFFSET,
+  ARRAY_PAIR_LIST,
+  ARRAY_PAIR,
+  EXIT_EXPR,
+  LIST_EXPR,
+  LIST_ASSIGNMENT_EXPR,
+  ASSIGNMENT_LIST_ELEMENT,
+  ASSIGNMENT_LIST,
+  FUNCTION_EXPRESSION,
+  ENCAPS_STRING_LITERAL,
+  COMPLEX_ENCAPS_VARIABLE,
+  SEMI_COMPLEX_ENCAPS_VARIABLE,
+  SEMI_COMPLEX_RECOVERY_EXPRESSION,
+  SIMPLE_ENCAPS_VARIABLE,
+  ENCAPS_LIST,
+  ENCAPS_VAR_OFFSET,
+  ENCAPS_VAR,
+  ENCAPS_DIMENSIONAL_OFFSET,
+  ENCAPS_OBJECT_MEMBER_ACCESS,
+
+  EXPRESSION;
+
+
+  public static LexerlessGrammar createGrammar() {
+    return createGrammarBuilder().build();
+  }
+
+  public static LexerlessGrammarBuilder createGrammarBuilder() {
+    LexerlessGrammarBuilder b = LexerlessGrammarBuilder.create();
+
+//    b.rule(COMPILATION_UNIT).is(b.optional(SCRIPT), SPACING, EOF);
+    // fixme TOP_STATEMENT_LIST instead of EXPRESSION
+    b.rule(SCRIPT).is(FILE_OPENING_TAG, b.optional(EXPRESSION));
+
+    lexical(b);
+    punctuators(b);
+    keywords(b);
+
+    // FIXME COMPILATION_UNIT
+    b.setRootRule(EXPRESSION);
+
+    return b;
+  }
+
+  public static void lexical(LexerlessGrammarBuilder b) {
+    b.rule(SPACING).is(
+      b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+")),
+      b.zeroOrMore(
+        b.commentTrivia(b.regexp(LexicalConstant.COMMENT)),
+        b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+")))
+    ).skip();
+
+    // Literals
+    b.rule(HEREDOC).is(SPACING, b.regexp(LexicalConstant.HEREDOC));
+    b.rule(NUMERIC_LITERAL).is(SPACING, b.regexp(LexicalConstant.NUMERIC_LITERAL));
+//    b.rule(STRING_LITERAL).is(SPACING, b.firstOf(b.regexp(LexicalConstant.STRING_LITERAL), ENCAPS_STRING_LITERAL));
+
+    b.rule(STRING_WITH_ENCAPS_VAR_CHARACTERS).is(b.regexp(LexicalConstant.STRING_WITH_ENCAPS_VAR_CHARACTERS));
+//    b.rule(ENCAPS_STRING_LITERAL).is(SPACING, "\"", ENCAPS_LIST, "\"");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)